helicone-node/async_logger/HeliconeAsyncOpenAI.ts

```diff
@@ -0,0 +1,203 @@
+import { Helicone } from "../core/HeliconeOpenAIApi";
+import {
+  HeliconeAsyncLogger,
+  HeliconeAsyncLogRequest,
+  Provider,
+  ProviderRequest,
+} from "./HeliconeAsyncLogger";
+import OpenAI from "openai";
+import * as Core from "openai/core";
+import { ChatCompletionCreateParams } from "openai/resources/chat/completions";
+import { CompletionCreateParams } from "openai/resources/completions";
+import * as hcone from "./HeliconeOpenAITypes";
+import { HeliconeHeaderBuilder } from "../core/HeliconeHeaderBuilder";
+import { IHeliconeAsyncClientOptions } from "../core/HeliconeClientOptions";
+import { Stream } from "openai/streaming";
+import { EmbeddingCreateParams } from "openai/resources";
+import { wrapAPIPromise } from "./HeliconeMonkeyPatch";
+import { Response } from "@whatwg-node/fetch";
+
+export class HeliconeAsyncOpenAI extends OpenAI {
+  public helicone: Helicone;
+  private originalCompletionCreate: hcone.CompletionCreateFunction;
+  private originalChatCreate: hcone.ChatCreateFunction;
+  private originalEmbeddingCreate: hcone.EmbeddingCreateFunction;
+  private heliconeHeaders: { [key: string]: string };
+  private logger: HeliconeAsyncLogger;
+
+  constructor(private options: IHeliconeAsyncClientOptions) {
+    const {
+      apiKey = Core.readEnv("OPENAI_API_KEY"),
+      organization = Core.readEnv("OPENAI_ORG_ID") ?? null,
```

Comment:
do we always have to pull this out? Is it possible to inherit this behavior from somewhere?